### ISSUE TITLE
URN Support

### ISSUE THREAD

**Issue by mtiller-jh** [JuliaWeb/URIs.jl#57]
I just want to point out that RFC 3986 covers both URLs and URNs under the umbrella of URIs.  But there is no mention of URNs anywhere in this project.  Granted, it is possible to parse a URN as a URI.  But the data types still have fields that are URL specific (_e.g.,_ `host`, `path`) which are URL specific and don't contain any of the terms associated with URNs from [RFC 8141](https://datatracker.ietf.org/doc/html/rfc8141) (_e.g.,_ `NID` or `NSS`).

Also, the `isvalid` function isn't actually performing the checks from RFC 8141 relating to valid URNs.  The result is that invalid URNs (_e.g.,_ `urn:--name:foo`) are considered valid by the `isvalid` function and valid URNs (_e.g.,_ `urn:example:weather?=op=map&lat=39.56&lon=-104.85&datetime=1969-07-21T02:56:15Z`) are considered invalid by the `isvalid` function.

In short, I think in order to really claim RFC 3986 compatibility, some work needs to be done to add support for RFC 8141 _or_ the documentation should be updated to reflect that the library is really meant for supporting only URLs.

Thoughts?

**Comment by quinnj** [comment]
Generic URN strings already parse as scheme `urn` plus an NID/NSS path. However, `isvalid(::URI)` still uses an obsolete fixed scheme allowlist and cannot provide RFC 8141 namespace semantics. Keeping this open for 2.0 work: generic validity should follow RFC 3986, with optional URN semantic validation kept separate.

[investigated by codex; reviewed by quinnj]
